profiling-ffi/src/exporter.rs

```diff
@@ -183,6 +184,7 @@ pub unsafe extern "C" fn ddog_prof_Exporter_Request_build(
     files: Slice<File>,
     additional_tags: Option<&ddcommon_ffi::Vec<Tag>>,
     endpoints_stats: Option<&profiled_endpoints::ProfiledEndpointsStats>,
+    metrics: Option<&ddcommon_ffi::Vec<Metric>>,
```

Comment:
Why not send it as part of the `files` slice?

Reply:
can you elaborate on this please ?

From what I understand, it would mean that we will duplicate `code` in different profilers (creating a file with the metric, adding it to the `files` slice...) for `metrics.json` file which is not specific to a profiler (I mean, profilers may or may not send metrics, but this json file is a `standard` defined in the backend that profilers could take advantage on).

Reply:
Another thing using the `files` slice, it's not possible to give a `content-type`.

Reply:
I guess there's a trade-off here, that may be different for every language.

For instance, for Ruby, I think it's actually going to be roughly as much code between using Ruby to generate JSON, or using the proposed API (perhaps even a bit more C code for this API). So it's not _that_ useful for Ruby versus Ruby directly emitting the `json` and including it in the files -- I'd probably go with that approach instead of using this.

But if it's useful for .net, since the complexity is also low, I don't see why not -- it's not like we can't change our minds later and deprecate this if we really want to.